rust/src/credential/mdoc.rs

```diff
@@ -223,3 +224,54 @@ pub enum MdocEncodingError {
     #[error("failed to encode Document to CBOR")]
     DocumentCborEncoding,
 }
+
+/// Convert a ciborium value to a serde_json value for display.
+fn to_json_for_display(value: &ciborium::Value) -> Option<serde_json::Value> {
+    /// Convert integer and text keys to strings for display.
+    fn key_to_string_for_display(value: &ciborium::Value) -> Option<String> {
+        match value {
+            ciborium::Value::Integer(i) => Some(<i128>::from(*i).to_string()),
+            ciborium::Value::Text(s) => Some(s.clone()),
+            ciborium::Value::Float(f) => Some(f.to_string()),
+            ciborium::Value::Bool(b) => Some(b.to_string()),
+            ciborium::Value::Null => Some("null".to_string()),
+            ciborium::Value::Tag(_, v) => key_to_string_for_display(v),
+            _ => {
+                tracing::warn!("unsupported key type: {:?}", value);
+                None
+            }
+        }
+    }
+
+    match value {
+        ciborium::Value::Integer(i) => Some(serde_json::Value::Number(i128::from(*i).into())),
+        ciborium::Value::Text(s) => Some(serde_json::Value::String(s.clone())),
+        ciborium::Value::Array(a) => Some(serde_json::Value::Array(
+            a.iter().filter_map(to_json_for_display).collect::<Vec<_>>(),
+        )),
+        ciborium::Value::Map(m) => Some(serde_json::Value::Object(
+            m.iter()
+                .filter_map(|(k, v)| {
+                    let key = key_to_string_for_display(k)?;
+                    let value = to_json_for_display(v)?;
+                    Some((key, value))
+                })
+                .collect(),
+        )),
+        ciborium::Value::Bytes(items) => Some(format!("0x{}", hex::encode(items)).into()),
+        ciborium::Value::Float(f) => {
+            let Some(num) = serde_json::Number::from_f64(*f) else {
+                tracing::warn!("failed to convert float to number: {}", f);
+                return None;
+            };
+            Some(serde_json::Value::Number(num))
+        }
+        ciborium::Value::Bool(b) => Some(serde_json::Value::Bool(*b)),
+        ciborium::Value::Null => Some(serde_json::Value::Null),
+        ciborium::Value::Tag(_, value) => to_json_for_display(value),
+        _ => {
+            tracing::warn!("unsupported value type: {:?}", value);
+            None
+        }
+    }
```

Comment:
Worth moving these to `isomdl` crate?

Reply:
I'll keep it here for now, it's pretty specific to the wallet unless we think it's useful in verifier code?